packages/web-components/src/tabs/tabs.ts

```diff
@@ -0,0 +1,194 @@
+    } else return this.previousActiveTabData.height / tab.getBoundingClientRect().height;
+  }
+
+  /**
+   * clearAnimationProperties
+   * resets animation values to defaults. removes the animated css class
+   */
+  private clearAnimationProperties(tab: Tab) {
+    this.previousActiveTabData = { id: '', x: 0, y: 0, height: 0, width: 0 };
+    this.activeTabOffset = 0;
+    this.activeTabScale = 1;
+    tab.classList.remove('animated');
+  }
+
+  /**
+   * animationLoop
+   * runs through all the operations required for setting the tab active indicator to its starting location, ending location, and applying the animated css class to the tab.
+   * @param tab
+   */
+  private animationLoop(tab: Tab) {
+    // If selected, set the animation properties.
+    if (tab.ariaSelected && tab.ariaSelected === 'true') {
+      // get the offset for active indicator
+      // the offset at the start of the animation will be at the location of the previously selected tab
+      this.calculateAnimationProperties(tab);
+      // set the active indicator offset and scale of the active indicator to css vars
+      this.setTabScaleCSSVar();
+      this.setTabOffsetCSSVar();
+
+      // clear properties for next animation
+      this.clearAnimationProperties(tab);
+    }
+
+    this.previousActiveTabData = this.activeTabData;
+    this.calculateAnimationProperties(tab);
+
+    // add the animate css class if the calculated offset is 0 and scale is 1
+    if (this.activeTabOffset === 0 && this.activeTabScale === 1) {
+      tab.classList.add('animated');
+      // update the css vars
+      this.setTabScaleCSSVar();
+      this.setTabOffsetCSSVar();
+      // now when the css class is added the active indicator ::after element will animate
+      // from the previous tab location to its starting location
+    }
+  }
+
+  /**
+   * setTabData
+   * sets the data from the active tab onto the class. used for making all the animation calculations for the active tab indicator.
+   */
+  private setTabData(): void {
+    if (this.tabs && this.tabs.length > 0) {
+      const tabs = this.tabs as Tab[];
+      const activeTab = this.activetab || tabs[0];
+      const activeRect = activeTab?.getBoundingClientRect();
+      const parentRect = this.getBoundingClientRect();
+
+      this.activeTabData = {
+        id: activeTab.id,
+        x: activeRect.x - parentRect.x,
+        y: activeRect.y - parentRect.y,
+        height: activeRect.height,
+        width: activeRect.width,
+      } as TabData;
+
+      if (!this.previousActiveTabData?.id) {
+        this.previousActiveTabData = this.activeTabData;
+      }
+    }
+  }
+
+  private setTabOffsetCSSVar() {
+    document.documentElement.style.setProperty(TabTokenNames.tabIndicatorOffset, `${this.activeTabOffset}px`);
+  }
+
+  private setTabScaleCSSVar(newScale?: number) {
+    document.documentElement.style.setProperty(TabTokenNames.tabIndicatorScale, `${newScale || this.activeTabScale}`);
+  }
```

Comment:
This is never called with a newScale value. Is that param necessary at all?

Reply:
oops!